core/src/test/scala/unit/kafka/coordinator/GroupMetadataManagerTest.scala

```diff
@@ -86,6 +88,263 @@ class GroupMetadataManagerTest {
+    EasyMock.expect(fileRecordsMock.readInto(EasyMock.anyObject(classOf[ByteBuffer]), EasyMock.anyInt()))
+      .andReturn(records.buffer)
+
+    EasyMock.replay(logMock, replicaManager, fileRecordsMock)
+
+    groupMetadataManager.loadGroupsAndOffsets(offsetTopicPartition, _ => ())
+
+    val loadedGroup = groupMetadataManager.getGroup(groupId)
+    loadedGroup match {
+      case Some(group) =>
+        assertEquals(groupId, group.groupId)
+        assertEquals(Empty, group.currentState)
+        assertEquals(committedOffsets.size - 1, group.allOffsets.size)
+        committedOffsets.foreach { case (topicPartition, offset) =>
+          if (topicPartition == tombstonePartition)
+            assertEquals(None, group.offset(topicPartition))
+          else
+            assertEquals(Some(offset), group.offset(topicPartition).map(_.offset))
+        }
+      case None =>
+        fail("Group was not loaded into the cache")
+    }
+  }
+
+  @Test
+  def testLoadOffsetsAndGroup() {
+    val offsetTopicPartition = new TopicPartition(Topic.GroupMetadataTopicName, groupPartitionId)
+    val logMock =  EasyMock.mock(classOf[Log])
+    val fileRecordsMock = EasyMock.mock(classOf[FileRecords])
+    val startOffset = 15L
+    val endOffset = 18L
+
+    val committedOffsets = Map(
+      new TopicPartition("foo", 0) -> 23L,
+      new TopicPartition("foo", 1) -> 455L,
+      new TopicPartition("bar", 0) -> 8992L
+    )
+    val offsetCommitRecords = committedOffsets.map { case (topicPartition, offset) =>
+      val offsetAndMetadata = OffsetAndMetadata(offset)
+      val offsetCommitKey = GroupMetadataManager.offsetCommitKey(groupId, topicPartition)
+      val offsetCommitValue = GroupMetadataManager.offsetCommitValue(offsetAndMetadata)
+      Record.create(offsetCommitKey, offsetCommitValue)
+    }.toSeq
+
+    val memberId = "98098230493"
+    val groupMetadataRecord = {
+      val group = buildStableGroupWithMember(memberId)
+      val groupMetadataKey = GroupMetadataManager.groupMetadataKey(groupId)
+      val groupMetadataValue = GroupMetadataManager.groupMetadataValue(group, Map(memberId -> Array.empty[Byte]))
+      Record.create(groupMetadataKey, groupMetadataValue)
+    }
+
+    val records = MemoryRecords.withRecords(startOffset, offsetCommitRecords ++ Seq(groupMetadataRecord): _*)
+
+    EasyMock.expect(replicaManager.getLog(offsetTopicPartition)).andStubReturn(Some(logMock))
+    EasyMock.expect(logMock.logStartOffset).andStubReturn(startOffset)
+    EasyMock.expect(replicaManager.getHighWatermark(offsetTopicPartition)).andStubReturn(Some(endOffset))
+    EasyMock.expect(logMock.read(EasyMock.eq(startOffset), EasyMock.anyInt(), EasyMock.eq(None), EasyMock.eq(true)))
+      .andReturn(FetchDataInfo(LogOffsetMetadata(startOffset), fileRecordsMock))
+    EasyMock.expect(fileRecordsMock.readInto(EasyMock.anyObject(classOf[ByteBuffer]), EasyMock.anyInt()))
+      .andReturn(records.buffer)
```

Comment:
The `expect` calls and the `committedOffsets` code seems similar in various tests. Would it make sense to extract one method for each? The former seems like more of a win, if possible. But the latter could replace 6 lines with 1.